x/ibctransfermiddleware/keeper/msg_server.go

```diff
@@ -0,0 +1,38 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/notional-labs/composable/v6/x/ibctransfermiddleware/types"
+)
+
+var _ types.MsgServer = msgServer{}
+
+// msgServer is a wrapper of Keeper.
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the x/ibctransfermiddleware MsgServer interface.
+func NewMsgServerImpl(k Keeper) types.MsgServer {
+	return &msgServer{
+		Keeper: k,
+	}
+}
+
+// UpdateParams updates the params.
+func (ms msgServer) UpdateCustomIbcParams(goCtx context.Context, req *types.MsgUpdateCustomIbcParams) (*types.MsgUpdateParamsCustomIbcResponse, error) {
+	if ms.authority != req.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.authority, req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := ms.SetParams(ctx, req.Params); err != nil {
```

Comment:
should we add logic to add params channel by channel?

Reply:
Yes. 100%
we need it..
this code is from the original

Reply:
no.
we will use gov to provide new settings.